src/main/scala/scalaz/reactive/Event.scala

```diff
@@ -1,11 +1,37 @@
 package scalaz.reactive
+import scalaz.Monoid
+import scalaz.zio.{ IO, RTS }
 
-case class Event[+A](value: Future[Reactive[A]]) extends AnyVal {
+case class Event[A](value: Future[Reactive[A]]) extends RTS { self =>
+
+  def merge(
+    v: Event[A]
+  ): Event[A] = { // if Event[+A]: covariant type A occurs in contravariant position
+
+    val winner: IO[Nothing, Either[(Time, Reactive[A]), (Time, Reactive[A])]] =
+      self.value.force.map(Left(_)).race(v.value.force.map(Right(_)))
```

Comment:
There is no need to put them inside an Either, the resulting IO will have the winner, so `self.value.force race v.value.force` will be enough

Reply:
True - at one moment I had the idea I needed the loser too, but that is not the case.